### PR TITLE
test suite: run osu micro benchmarks

### PR DESCRIPTION
**Pull request by garlick**
This pulls in a copy of the [OSU micro benchmarks](https://mvapich.cse.ohio-state.edu/benchmarks/) and adds a sharness test to run an assortment of them in CI.

This is probably overkill, but I just wanted to be sure that MPI bootstrapped with our PMIx server is fully functional.

The test does take a long time.  I culled it down a bit so it takes about 40s to run the sharness script on my desktop.  Hopefully that will be OK in CI.

### CODE REVIEW

**Comment by garlick** [comment]
I underestimated how long this would take to run in CI!  Culled some more tests from the "normal" path and it's down from 40s to 15s on my desktop.

**Comment by garlick** [comment]
Well that was slightly ridiculous.  It seems practically none of these tests can run in CI without taking too long.  I cut it back to just the init tests.  Oh well, we can still run them from time to time with `TEST_LONG=t`.

I added in a couple more things - suppressing a warning error from every rank for an "experimental" pmix attribute, tightening up attribute type checking, and setting the module name for future shell logging enhancement.

**Comment by garlick** [comment]
Repushed to fix the shell logging patch, which I had applied to the wrong Makefile.am!  Tested with flux-framework/flux-core#3879

File: src/shell/plugins/Makefile.am [hunk 1]
```diff
@@ -4,7 +4,9 @@ AM_CFLAGS = \
 AM_LDFLAGS = \
 	$(CODE_COVERAGE_LIBS)
 AM_CPPFLAGS = \
-	-I$(top_srcdir)
```
Comment (grondo) [inline src/shell/plugins/Makefile.am:7]: Commit message typo: `wtih`

Also, something to be aware of is that the new `flux/shell.h` changes _require_ that `FLUX_SHELL_PLUGIN_NAME` be set before including `shell.h` (The commit message here makes it sound like it is optional, which maybe it should be?)

**Review by grondo — APPROVED** [review]
LGTM! Just one commit message typo I caught

File: src/shell/plugins/Makefile.am (hunk 1 above)
Comment (garlick) [reply src/shell/plugins/Makefile.am:7]: Thanks, I reworded that commit to:
```
build: define FLUX_SHELL_PLUGIN_NAME
    
Problem: an upcoming change to shell logging will require
FLUX_SHELL_PLUGIN_NAME to be defined before including shell.h.
    
Add definition to AM_CPPFLAGS.
```

**Pull request merged**